torch/nested/_internal/nested_tensor.py

```diff
@@ -73,34 +73,31 @@ class NestedTensor(torch.Tensor):
     _strides: Tuple[int, ...]
     # Indicates that the nth dimension is ragged
     _ragged_idx: int
-    _metadata_cache: Dict[str, Any]
 
     @staticmethod
     def __new__(
         cls,
-        values,
-        offsets,
-        *,
-        lengths=None,
+        values: torch.Tensor,
+        metadata: torch.Tensor,  # This will be a subclass representing raggedness
+        non_contig_offsets: Optional[torch.Tensor] = None,
         **kwargs,
```

Comment:
`ragged_idx` is still handled through kwargs, right?

Reply:
Yes